Core/GDCore/Project/Project.cpp

```diff
@@ -642,6 +683,29 @@ void Project::UnserializeFrom(const SerializerElement& element) {
   if (currentPlatform == NULL && !platforms.empty())
     currentPlatform = platforms.back();
 
+  // TODO EBO All events based behaviors and events based objects should be
+  // added to the project before the content of events based objects is
+  // unserialized because when unserializing child-objects InsertNewObject and

```

Comment:
Might be a good reason for storing separately the data of an object (something like "gd::BehaviorContent") from the code manipulating it (something like "gd::Behavior") (noting that the naming is not perfect)

Reply:
I could probably work but Behavior can use the same logic as CustomObject and it gives better encapsulation.
I guess that BehaviorContent was used because the implementation was JS side.

I think that a solution would be to unserialize only empty EventsBasedObjet in a 1st pass and then their content. I have a PR ready for it when this one is merged.